engine-rest/engine-rest/src/test/java/org/camunda/bpm/engine/rest/ProcessInstanceRestServiceInteractionTest.java

```diff
@@ -4551,4 +4554,75 @@ private void verifyTaskComments(List<Comment> mockTaskComments, Response respons
   }
+
+  @Test
+  public void testWithVariablesInReturnSetToFalseGetProcessList() {
+    List<ProcessInstance> mockProcessInstanceList = new ArrayList<>();
+    ProcessInstance mockInstance = MockProvider.createMockInstance();
+    mockProcessInstanceList.add(mockInstance);
+		
+    ProcessInstanceQuery sampleInstanceQuery = mock(ProcessInstanceQuery.class);
+			
+    MockedStatic<QueryUtil> mockedStatic = mockStatic(QueryUtil.class);
+    Mockito.when(QueryUtil.list(sampleInstanceQuery, 0, 1)).thenReturn(mockProcessInstanceList);
+    assertEquals(1, QueryUtil.list(sampleInstanceQuery, 0, 1).size());
+
+    when(runtimeServiceMock.createProcessInstanceQuery()).thenReturn(sampleInstanceQuery);
+    when(sampleInstanceQuery.processInstanceId(anyString())).thenReturn(sampleInstanceQuery);
+    when(sampleInstanceQuery.listPage(0, 1)).thenReturn(mockProcessInstanceList);
+	         
+    VariableMap mockInstanceWithVariable = MockProvider.createMockSerializedVariables();
+    when(runtimeServiceMock.getVariables(anyString())).thenReturn(mockInstanceWithVariable);
+
+    given()
+      .queryParam("withVariablesInReturn", false)
+      .queryParam("firstResult", 0) .queryParam("maxResults", 1)
+    .then().expect().statusCode(Status.OK.getStatusCode())
+      .contentType(MediaType.APPLICATION_JSON)
+      .body("[0].id", Matchers.equalTo(MockProvider.EXAMPLE_PROCESS_INSTANCE_ID))
+      .body("[0].definitionId", Matchers.equalTo(MockProvider.EXAMPLE_PROCESS_DEFINITION_ID))
+      .body("[0].businessKey", Matchers.equalTo(MockProvider.EXAMPLE_PROCESS_INSTANCE_BUSINESS_KEY))
+      .body("[0].suspended", Matchers.equalTo(MockProvider.EXAMPLE_PROCESS_INSTANCE_IS_SUSPENDED))
+      .body("[0].tenantId", Matchers.equalTo(MockProvider.EXAMPLE_TENANT_ID))
+      .body("[0].variables",  Matchers.nullValue())
+    .when().get(PROCESS_INSTANCE_URL);
+	    
+    mockedStatic.close();
+
+  }
+  
+  @Test
+  public void testWithVariablesInReturnSetToTrueGetProcessList() {
+    List<ProcessInstance> mockProcessInstanceList = new ArrayList<>();
+    ProcessInstance mockInstance = MockProvider.createMockInstance();
+    mockProcessInstanceList.add(mockInstance);
+	
+    ProcessInstanceQuery sampleInstanceQuery = mock(ProcessInstanceQuery.class);
+		
+    MockedStatic<QueryUtil> mockedStatic = mockStatic(QueryUtil.class);
+    Mockito.when(QueryUtil.list(sampleInstanceQuery, 0, 1)).thenReturn(mockProcessInstanceList);
+    assertEquals(1, QueryUtil.list(sampleInstanceQuery, 0, 1).size());
+
+    when(runtimeServiceMock.createProcessInstanceQuery()).thenReturn(sampleInstanceQuery);
+    when(sampleInstanceQuery.processInstanceId(anyString())).thenReturn(sampleInstanceQuery);
+    when(sampleInstanceQuery.listPage(0, 1)).thenReturn(mockProcessInstanceList);
+         
+    VariableMap mockInstanceWithVariable = MockProvider.createMockSerializedVariables();
+    when(runtimeServiceMock.getVariables(anyString())).thenReturn(mockInstanceWithVariable);
```

Comment:
I think it would be better to check if getVariables is called with the actual processInstanceId and not any strig.